### PR TITLE
Optimize: reduce convertion on previous k8s service

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
@ramaraochavali @dddddai

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -581,14 +581,8 @@ func (c *Controller) onServiceEvent(prev, curr any, event model.Event) error {
...
-	case model.EventAdd:
-		c.addOrUpdateService(nil, currSvc, svcConv, event, false)
-	case model.EventUpdate:
-		prevSvc := controllers.Extract[*v1.Service](prev)
-		if prevSvc == nil {
-			return nil
-		}
-		c.addOrUpdateService(prevSvc, currSvc, svcConv, event, false)
+	default:
+		c.addOrUpdateService(currSvc, svcConv, event, false)
```
[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:585]: How could we do? c.deleteService also need the converted svc

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 2]
```diff
@@ -663,11 +659,6 @@ func (c *Controller) addOrUpdateService(prev, curr *v1.Service, currConv *model.
 	}
 
 	c.opts.XDSUpdater.SvcUpdate(shard, string(currConv.Hostname), ns, event)
-	var prevConv *model.Service
-	if event == model.EventUpdate && prev != nil {
-		prevConv = kube.ConvertService(*prev, c.opts.DomainSuffix, c.Cluster())
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:668]: @howardjohn ptal, reduced one time conversion

**Comment by hzxuzhonghu** [comment]
> do we have any benchmarks of this code?

we do not have any benchmark on xds trigger anywhere in istio. And I do think it is not the hotpoint compared to xds generator

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**